File: leetCode/easy/283-moveZeroes.c
#include <stdio.h>

// 0 1 0 3 12
// 0 
// temp = 1
// nums[1] () = 0
// 1 0 0 3 12
//  
void moveZeroes(int *nums, int numsSize){
 int nonZero = 0;
 for(int i = 0; i < numsSize; i++){
  if(nums[i] != 0){
    int temp = nums[i]; 
    nums[i] = nums[nonZero];
    nums[nonZero] = temp;
    zonzero++;
  }
 }

}


int main(void){

  int nums[5];
  int numsSize = 5;

  moveZeroes(nums[0], numsSize);



  return 0;
}

File: leetCode/easy/283-moveZeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int left = 0;
        for(int right = 0; right < nums.length; right++){
            if(nums[left] == 0){
                while(nums[right] == 0 && right != nums.length - 1) {
                    right++;
                    }
                nums[left] = nums[right];
                nums[right] = 0;
            } 
            left++;
        }
    }
}
